Controllers/cart.js

```diff
@@ -0,0 +1,66 @@
+const Cart = require('../Database/cart');
+const _ = require('lodash');
+
+exports.getCartItems = async (req, res) => {
+	const { cart } = req;
+	try {
+		await cart.populate('cartItems.product').execPopulate();
+		res.json({ success: true, response: cart });
+	} catch (error) {
+		res.json({ success: false, response: error.message });
+	}
+};
+
+exports.addCartItems = async (req, res) => {
+	const { product } = req;
+	const { cart } = req;
+	try {
+		if (!cart.cartItems.id(product._id)) {
+			const updateCart = _.extend(cart, {
+				cartItems: _.concat(cart.cartItems, {
```

Comment:
I think, we can extract this out in a separate variable to make it more readable